apps/api/src/app/shared/cache/cache-service.spec.ts

```diff
@@ -37,7 +53,11 @@ export const CacheService = {
         return;
       },
       delByPattern(pattern?: string) {
-        data = {};
+        const preFixSuffixTuple = pattern.split('*');
+
+        for (const key in data) {
+          if (key.startsWith(preFixSuffixTuple[0]) && key.endsWith(preFixSuffixTuple[1])) delete data[key];
+        }
```

Comment:
Connected to what @LetItRock mentioned in other PR, I think it's better to mock `ioredis` and not the whole service. Because right now. the testing here is misses the point since you recreated the functionality in the tests. So the real one is not tested.

Reply:
I see what you mean, on the initial thought on this test module I wanted to test the interceptors and not the service but the reasoning on what you both are saying, and will look into mocking the ioredis